src/main/java/ch/njol/skript/expressions/ExprDateAgoLater.java

```diff
@@ -45,28 +45,41 @@
                 "%timespan% (later|(from|after) [the] [date] %-date%)");
     }
 
+    @Override
+    @SuppressWarnings({"unchecked", "null"})
+    public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+        timespan = (Expression<Timespan>) exprs[0];
+        date = (Expression<Date>) exprs[1];
+        ago = matchedPattern == 0;
+        return true;
+    }
+
     @SuppressWarnings("null")
     private Expression<Timespan> timespan;
     @Nullable
     private Expression<Date> date;
 
     private boolean ago;
 
-    @SuppressWarnings("null")
-    @Nullable
     @Override
+    @Nullable
+    @SuppressWarnings("null")
     protected Date[] get(Event e) {
+		if (this.date == null || this.timespan == null)
+			return null;
+
         Timespan timespan = this.timespan.getSingle(e);
-        Date date = this.date == null ? new Date() : this.date.getSingle(e);
-        if (timespan == null || date == null) {
-            return null;
-        }
-        if (ago) {
+		Date otherDate = this.date.getSingle(e);
+		if (timespan == null || otherDate == null)
+			return null;
+
+        Date date = this.date == null ? new Date() : new Date(otherDate.getTimestamp());
+        if (ago)
```

Comment:
You should use Date#plus and Date#minus, it'll remove the need to create a new date instance yourself too